packages/storybook/stories/MessageThread/snippets/MessageWithCustomAttachment.snippet.tsx

```diff
@@ -0,0 +1,103 @@
+import {
+  AttachmentMenuAction,
+  AttachmentMetadata,
+  ChatMessage,
+  FluentThemeProvider,
+  MessageStatus,
+  MessageThread,
+  defaultAttachmentMenuAction
+} from '@azure/communication-react';
+import { initializeFileTypeIcons } from '@fluentui/react-file-type-icons';
+import { ShareIos24Regular, WindowNew24Regular } from '@fluentui/react-icons';
+import React from 'react';
+
+initializeFileTypeIcons();
+
+export const MessageWithCustomAttachment: () => JSX.Element = () => {
+  return (
+    <FluentThemeProvider>
+      <MessageThread
+        userId={'1'}
+        showMessageDate={true}
+        attachmentOptions={{
+          downloadOptions: {
+            actionsForAttachment: (attachment: AttachmentMetadata, message?: ChatMessage): AttachmentMenuAction[] => {
+              // allows the menu actions to be dynamically generated based on the attachment content
+              // (i.e. name, extension, etc.), or the message content (i.e. senderID, etc.)
+              // in this example, we are constructing a dynamic menu based on the attachment extension
+              if (attachment.extension === 'pdf') {
+                return [
+                  {
+                    name: 'Share',
+                    icon: <ShareIos24Regular />,
+                    onClick: () => {
+                      return new Promise((resolve) => {
+                        window.open(attachment.url, '_blank');
+                        resolve();
+                      });
+                    }
+                  }
+                ];
+              } else {
+                return [
+                  defaultAttachmentMenuAction,
+                  {
+                    name: 'Open',
+                    icon: <WindowNew24Regular />,
+                    onClick: () => {
+                      return new Promise((resolve) => {
+                        window.alert('open button clicked for messageID: ' + message?.messageId);
+                        resolve();
+                      });
+                    }
+                  }
+                ];
+              }
+            }
+          }
+        }}
+        messages={[
+          {
+            messageType: 'chat',
+            senderId: 'user1',
+            senderDisplayName: 'Kat Larsson',
+            messageId: Math.random().toString(),
+            content: 'This is the report:',
+            createdOn: new Date('2019-04-13T00:00:00.000+08:10'),
+            mine: false,
+            attached: false,
+            status: 'seen' as MessageStatus,
+            contentType: 'text',
+            attachments: [
+              {
+                id: 'SomeUniqueId1',
+                name: 'Report.docx',
```

Comment:
Title for this attachment doen't look centered vertically. Is it expected?
![image](https://github.com/Azure/communication-ui-library/assets/98852890/71425fca-6570-442f-8cc9-7c634dc05bbe)

Reply:
this is a known issue that will be address in future PRs

Reply:
cc @jimchou-dev